hoomd/mpcd/DryingDropletStreamingMethod.h

```diff
@@ -0,0 +1,271 @@
+    const Scalar V = (end_R - start_R)/(m_period * m_mpcd_dt);
+    
+    if (V > 0)
+        {
+        throw std::runtime_error("Droplet radius must decrease.");
+        }
+    
+    /*
+     * If the initial geometry was not set. Set the geometry and validate the geometry 
+     * Because the interface is shrinking, it is sufficient to validate only the first time the geometry
+     * is set.
+     */
+    if (m_geometry_set && m_validate_geom)
+        {
+        m_geom = std::make_shared<mpcd::detail::SphereGeometry>(start_R, V , m_bc ); 
+        validate();
+        m_geometry_set = false;
+        m_validate_geom = false;
+        }
+    /*
+     *Update the geometry radius
+     */
+    m_geom = std::make_shared<mpcd::detail::SphereGeometry>(end_R, V, m_bc );
+    
+    //stream according to base class rules
+    ConfinedStreamingMethod<mpcd::detail::SphereGeometry>::stream(timestep);
+
+    //calculating number of particles to remove from solvent density
+    const unsigned int N_remove = round((4.*M_PI/3.)*end_R*end_R*end_R*m_density);
+    const int N_evap = m_mpcd_pdata->getNGlobal() - N_remove;
+    std::cout << "Number of particles to evaporate is : " << N_evap << std::endl;
+
+    // get the compact array of indexes of bounced particles
+    unsigned int N_bounced = 0;
+    const unsigned int N = m_mpcd_pdata->getN();
+
+    ArrayHandle<unsigned char> h_bounced(m_bounced, access_location::host, access_mode::read);
+
+    bool overflowed = false;
+    do
+        {
+        if (overflowed)
+            {
+            GPUVector<unsigned char> bounced_index(N_bounced, m_exec_conf);
+            m_bounced_index.swap(bounced_index);
+            // resize m_bounced_index with swap idiom
+            }
+        ArrayHandle<unsigned char> h_bounced_index(m_bounced_index, access_location::host, access_mode::overwrite);
+        const unsigned int N_bounced_max = m_bounced_index.size();
+        N_bounced = 0;
+        for (unsigned int idx=0; idx < N; ++idx)
+            {
+            if (h_bounced.data[idx])
+                {
+                if (N_bounced < N_bounced_max)
+                    {
+                    h_bounced_index.data[N_bounced] = idx;
+                    }
+                ++N_bounced;
+                }
+            }
+        overflowed = N_bounced > N_bounced_max;
+        } while (overflowed);
+    std::cout << "N_bounced is : " << N_bounced << std::endl;
+
+    // reduce / scan the number of particles that were bounced on all ranks
+    unsigned int N_bounced_total = N_bounced;
+    unsigned int N_before = 0;
+    #ifdef ENABLE_MPI
+    if (m_exec_conf->getNRanks() > 1)
+        {
+        MPI_Allreduce(&N_bounced, &N_bounced_total, 1, MPI_UNSIGNED, MPI_SUM, m_exec_conf->getMPICommunicator());
+        MPI_Exscan(&N_bounced, &N_before, 1, MPI_UNSIGNED, MPI_SUM, m_exec_conf->getMPICommunicator());
+        }
+    #endif // ENABLE_MPI
+
+    // N_evap is the total number to remove (computed previously)
```

Comment:
The code that follows will all get changed to implement the deletion logic, so I am not going to look at that now